libretroshare/src/rsitems/rsphotoitems.cc

```diff
@@ -55,25 +55,23 @@ void RsGxsPhotoAlbumItem::serial_process(RsGenericSerializer::SerializeJob j,RsG
      RsTypeSerializer::serial_process(j,ctx,TLV_TYPE_STR_NAME,     album.mPhotographer,  "mPhotographer");
      RsTypeSerializer::serial_process(j,ctx,TLV_TYPE_STR_DATE,     album.mWhen,          "mWhen");
      RsTypeSerializer::serial_process(j,ctx,TLV_TYPE_STR_LOCATION, album.mWhere,         "mWhere");
-     RsTypeSerializer::serial_process(j,ctx,TLV_TYPE_STR_PIC_TYPE, album.mThumbnail.type,"mThumbnail.type");
 
-	 RsTlvBinaryDataRef b(RS_SERVICE_GXS_TYPE_PHOTO, album.mThumbnail.data, album.mThumbnail.size);
-	 RsTypeSerializer::serial_process<RsTlvItem>(j, ctx, b, "thumbnail binary data") ;
+    RsTlvBinaryDataRef b(RS_SERVICE_GXS_TYPE_PHOTO, album.mThumbnail.mData, album.mThumbnail.mSize);
```

Comment:
Because RsGxsImage is a RsSerializable, you can directly call:

    album.mThumbnail.serial_process(j,ctx);

Reply:
RsPhotoAlbum should be made an RsSerializable itself and handle it's member serialization inside of it with the `RS_SERIAL_PROCESS(memberName)` macro

Reply:
I've fixed up the serialisation according to cyril's suggestion.
I'm happy to add serialisation to RsPhotoAlbum etc in another PR, I just want to get the basics working for now.
I looked through the other GxsItems - and none of them implement serialisation like this. Is there a reason why?

Reply:
New items should use the macro RS_SERIAL_PROCESS. It's the simplest. The other older items (that havn't been converted yet) still explicitely call serial_process().